src/TheThingsNetwork.cpp

```diff
@@ -202,6 +176,7 @@ bool TheThingsNetwork::join(int retries, long int retryDelay) {
 }
 
 bool TheThingsNetwork::join(const byte appEui[8], const byte appKey[16], int retries, long int retryDelay) {
+  configure_channels();
   reset(); 
```

Comment:
Shouldn't we call `configure_channels()` after `reset()`?

Also, I think `configure_channels()` has to be called once. Please discuss the user flow with @FokkeZB. There are imo different flows here: personalize, personalize with keys, join, join with keys and provision. If you call `configure_channels()` using the defaults, the user has no chance to set the spreading factor and FSB when using OTAA.